Done

XtratoS opened the pull request:
Pulling for the Reactions Bot

XtratoS committed:
Added the required basic functionality

Added the required basic functionality

.

XtratoS committed:
Done

Added all the requirements and some bonus commands

XtratoS committed:
.

XtratoS committed:
Added new field conf.example.js

Added a new field for the mongodb connection.

XtratoS committed:
Update commands.js

Using a more neat layout for module.export, add 1 line for a new command ( was 2 )

XtratoS committed:
Update commands.js

Using a more neat layout for module.export, add 1 line for a new command ( was 2 )

XtratoS committed:
Removed credentials

Removed credentials, now using the config file for the mongodb connection link.

XtratoS committed:
Removed credentials

Removed credentials, now using the config file for the mongodb connection link.

XtratoS committed:
Removed credentials

Removed credentials, now using the configuration file for the mongodb connection link

XtratoS committed:
Removed credentials

Removed credentials, now using the configuration file for the mongodb connection link

XtratoS committed:
Removed credentials

Removed credentials and added a `message link` field to the command

XtratoS committed:
Removed credentials

Removed credentials and added a `message link` field to the command

tyzoid commented on bot.js:
```diff
@@ -10,13 +10,23 @@ client.on('ready', function() {
 });
 
+// reaction roles code
+const reaction_bot = require("./reaction_roles.js")
```
Any requirements / includes should be at the top of the file.

tyzoid commented on bot.js:
```diff
@@ -10,13 +10,23 @@ client.on('ready', function() {
+    if ((event_type === "MESSAGE_REACTION_ADD" || event_type === "MESSAGE_REACTION_REMOVE")) {
+        reaction_bot.run(client, event).catch(err => console.log(err))
+	}
+})
```
This is fine for now, we'll want to move the discord things into it's own subfolder at a later date, though.

tyzoid commented on bot.js:
```diff
@@ -10,13 +10,23 @@ client.on('ready', function() {
+// reaction roles code
+const reaction_bot = require("./reaction_roles.js")
+// check every event for reaction add/remove on the reaction role message.
+client.on("raw", event => {
```
Unless necessary (i.e. to avoid rebinding of `this`) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

i.e.
```
var f = function(a, b) { return a + b; }
```

Can be re-written as:
```
var f = (a, b) => a+b;
```

But not:
```
var f = function(a, b) {
	if (a > b)
		return a - b;
	else
		return b - a;
}
```
```suggestion
client.on("raw", function(event) {
```

tyzoid commented on bot.js:
```diff
@@ -10,13 +10,23 @@ client.on('ready', function() {
 		var args = Args(msg.content);
 		var cmd = args[0].substring(conf.commandPrefix.length);
-
+		args.shift()
```
By convention, `args[0]` is always the command / program name. See `argv[]` on `main()` in most other languages, i.e. C, C++, Java, Python, etc.

Also, be sure to add a newline between control statements, i.e. before the `if` on line 28.
```suggestion

```

tyzoid commented on commands.js:
```diff
@@ -1,7 +1,10 @@
+	clearroles: require('./commands/clearroles.js'),
+	makemessage: require('./commands/makemessage.js'),
+	editmessage: require('./commands/editmessage.js')
+}
```
Let's line this up for now. In the future, I'd like to move away from a massive list of commands and instead look at publisher-subscriber for all events, but this isn't necessary for now.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+const Discord = require("discord.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
```
Let's steer clear of things like databases until we need them.

1. It introduces complexity where none is required.
2. Personally not a fan of mongodb, especially not for something as easily added to a relational DB.

For now, sticking with flatfile json files should be simple / easy.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
+const conf = require('../conf.js');
+mongoose.connect(conf.mongodbs, { useNewUrlParser: true })
```
Credentials still appear in other commits, we'll need to squash them together before approving.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+/**
+ * Adds a new role to any message in the server
+ */
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

```suggestion
module.exports = async function (bot, message, args) {
```

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+    channels = channels.first(channels.array().length)
+    // channels is now an arrray of all the channels in this server
+    let l = channels.length
+    let i
```
`let` and `var` are interspersed with no real benefit.

`let` is block-scoped, and `var` is function scoped. If both are in the root of the function, there's no difference. In general, you should use `var` unless `let` gives tangible benefits.

For example, using `let` here is no different from using `var`, since the scoping is the same.
```
var f = function() {
	var i = 0;
	let j = 4;
	while (i < j) {
		console.log(i++);
	}
}
```

Using `let` here is different from using `var`, since the scoping is within the `if` statement, however the impact is small since the object is small, so the benefit is tiny. In a large function, with large objects, `let` starts to make more sense. For consistency here, though, `var` is recommended.
```
var f = function(arg) {
	if (arg) {
		let j = {count: 1};
		arg += j.count;
	}

	return arg;
}
```

Additionally, always assign when declaring - doing one without the other is useless in a loosely-typed, loosely-scoped language like javascript.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+    let l = channels.length
+    let i
+    // now keep looping through the text channels and try to fetch the message
+    for (i = 0; i < l; i++) {
```
For example, drop the `let i` above, and change this to `for (var i = 0...`

```suggestion
    for (var i = 0; i < l; i++) {
```

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+            return await msg.edit(done).then(m => m.delete(30000))
+        }
+    })
+}
```
File should use tabs instead of spaces for indentation.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+    var role = message.mentions.roles.first()
+    var channel_id
+    // message placeholder
+    var msg
```
For related data, instead of making lots of variables, why not make one object?

```
var reactionrole = {
	messageid: args[1],
	emoji: args[2],
	role: message.mentions.roles.first(),
	/* etc */
}
```

Then, later you can reference through the new object, instead of via multiple variables. `reactionrole.messageid`, or `reactionrole.emoji`. This also makes it easier to serialize/deserialize via `JSON.stringify()` and `JSON.parse()`.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+            good_message = await channels[i].fetchMessage(message_id).catch(err => { })
+            // once message is found, react to it with the given emoji
+            if (good_message) {
+                await good_message.react(emoji).catch(err => { })
```
If you plan on squashing errors, why `catch` them at all here?

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+    // check if there's an emoji already bound to that message with a role.
+    reaction_message_model.findOne({
+        messageID: message_id
+    }, async (err, reaction_message) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

```suggestion
    }, async function (err, reaction_message) {
```

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+    }
+    // check if command issuer has perms
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+const Discord = require("discord.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
```
Move to flat file instead of jumping directly to a db.

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
+const conf = require('../conf.js');
+mongoose.connect(conf.mongodbs, { useNewUrlParser: true })
```
Same comment here with creds.

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Clears all the roles on any message in the server
+ */
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

```suggestion
module.exports = async function(bot, message, args) {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+    reaction_message_model.deleteMany({
+        guildID: message.guild.id
+    },(err, rem) => {
+        if(err) {console.log(err); return}
```
One-line `if` statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.

```suggestion
	if(err)
		return console.log(err);
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+        guildID: message.guild.id
+    },(err, rem) => {
+        if(err) {console.log(err); return}
+        if(!rem){
```
Spaces around control statements.

```suggestion
        if (!rem) {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+            .setColor("#F0FFF0")
+            .addField("Notification", "No roles were found in the database to remove.")
+            return message.channel.send(noq).then(r => r.delete(30000))
+        }else{
```
Spaces around control statements.
```suggestion
        } else {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+        }else{
+            let done = new Discord.RichEmbed()
+            .setColor("#00FFF0")
+            .addField("Notification", "All roles on all messages were deleted from the database.")
```
Line continuations should be indented. Also, let's use `var` instead of `let`. It's not really doing much for you here.

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+const Discord = require("discord.js")
+
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

```suggestion
module.exports = async function (bot, message, args) {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+        let help = new Discord.RichEmbed()
+        .setColor("#FFFF00")
+        .addField("Usage", "clearroles")
+        .addField("More info", "Clears all the roles on any message in the server.")
```
Chained functions on objects should be indented one level.

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+    if (args[0] == "help"){
+        let help = new Discord.RichEmbed()
+        .setColor("#FFFF00")
+        .addField("Usage", "editmessage <message_id> <message content>")
```
Chained functions should be indented one level.

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+    // put all channels in the server in an array
+    let channels = message.guild.channels
+    let l = channels.array().length
+    channels = channels.first(l)
```
Perhaps consolidating these into one object (as described above) would be helpful?

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+    let i
+    // now keep looping through the text channels and try to fetch the message
+    let good_message
+    message.channel.send("Loading...").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
    message.channel.send("Loading...").then(async function(m) {
```

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+                                    "Success",
+                                    `Message **${message_id}** have been edited.`
+                                )
+                        ).then(n=>n.delete(30000))
```
Spaces around operators / control statements.

```suggestion
                        ).then(n => n.delete(30000))
```

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+                            new Discord.RichEmbed()
+                                .setColor("#FF0000")
+                                .addField("Failture", `The bots isn't the author of this message hence the bot can't edit this message.`)
+                        ).then(m=>m.delete(60000))
```
Spaces around operators / control statements.
```suggestion
                        ).then(m => m.delete(60000))
```

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+            new Discord.RichEmbed()
+                .setColor("#FF0000")
+                .addField("Failture", `Couldn't find a message with the id **${message_id}** in this server.`)
+        ).then(m=>m.delete(60000))
```
Spaces around operators / control statements.
```suggestion
        ).then(m => m.delete(60000))
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+const Discord = require("discord.js")
+const hutil = require("../utilities/hutil.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
```
Move to flat file instead of jumping directly to a db.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+const hutil = require("../utilities/hutil.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
+mongoose.connect('mongodb+srv://XtratoS:UfTMXyIAxTN6Xr3q@nbot-dg4ad.mongodb.net/rbot?retryWrites=true', { useNewUrlParser: true })
```
Creds should not appear here.

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+const Discord = require("discord.js")
+
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
module.exports = async function(bot, message, args) {
```

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+        return message.channel.send(help)
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+            return m.delete(15000)
+        })
+    }
+    channel = message.guild.channels.find(f=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

```suggestion
    channel = message.guild.channels.find(function(f) {
```

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+    channel = message.guild.channels.find(f=>{
+        if(f.id == args[0])
+            return f
+    })
```
What do you return here if `f.id != args[0]`? What's supposed to be returned? I doubt `undefined` is an expected return value here.

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+    msg = message.toString().split(/ +/g).slice(2).join(" ")
+    if(msg.startsWith("image")){
+        msg = msg.toString().split(/ +/g).slice(1).join(" ")
+        return channel.send(msg.toString().split(/ +/g).slice(1).join(" "), {files: [msg.toString().split(/ +/g).slice(0,1).join(" ")]}).then(m=>{
```
Why are we messing around with splitting up and recombining `message` when `args` is already split up?

tyzoid commented on commands/makemessage.js:
```diff
@@ -0,0 +1,31 @@
+            message.delete()
+        })
+    }else{
+        return channel.send(msg).then(m=>{
```
Spaces around operators / control statements.
```suggestion
        return channel.send(msg).then(m => {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+ * -If a role/emoji is provided, removed that role/emoji from the specified message,
+ * -If no role/emoji provdided, removes all the roles/emojis from the specified message.
+ */
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
module.exports = async function(bot, message, args) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+            .addField("More info", "*Has 2 usages*:\nIf a role/emoji is provided, removed that role/emoji from the specified message,\n-If no role/emoji provdided, removes all the roles/emojis from the specified message.")
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
```
Let's put newlines between control blocks.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        })
+    }
+    var message_id = args[0]
+    if (!args[1]) {
```
Let's put newlines between control blocks.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+            .setDescription("*This command will be automatically cancelled in 60 seconds*")
+        msg = message.channel.send(removeall)
+        const filter = m => m.author.id === message.author.id
+        var response = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        var response = await message.channel.awaitMessages(filter, {max: 1, time: 60000}).then(async function(collected) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                ).then(m => m.delete(30000))
+                return false
+            }
+            if (collected.first().content.toLowerCase() == "yes") {
```
Let's put newlines between control blocks.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                return false
+            }
+        }).catch(r => { console.log(r) })
+        if (!response) {
```
Let's put newlines between control blocks.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        if (!response) {
+            return
+        }
+        reaction_message_model.deleteOne({
```
Let's put newlines between control blocks.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        }
+        reaction_message_model.deleteOne({
+            messageID: message_id
+        }, (err, rem) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        }, function(err, rem) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        reaction_message_model.deleteOne({
+            messageID: message_id
+        }, (err, rem) => {
+            if (err) { console.log(err); return }
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
		if (err)
			return console.log(err);
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        // remove only the specified emoji or corresponding role
+        reaction_message_model.findOne({
+            messageID: message_id
+        }, async (err, rem) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        }, async function(err, rem) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+        reaction_message_model.findOne({
+            messageID: message_id
+        }, async (err, rem) => {
+            if (err) { console.log(err); return }
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
		if (err) 
			return console.log(err);
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                            })
+                        } else {
+                            // if not empty just save the object after removing the role from it
+                            reaction_message_model.updateOne({ messageID: message_id }, { $set: { roles: rem.roles } }).catch(err => {
```
This is a long lone, perhaps re-format?
```suggestion
						reaction_message_model
							.updateOne({ messageID: message_id }, { $set: { roles: rem.roles } })
							.catch(function(err) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                    }
+                }
+                if (!success) {
+                    let failture = new Discord.RichEmbed()
```
`failure` is misspelled.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+            if (!collected.first()) {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FF0000")
+                    .addField("Failture", "Command timed out")
```
`failure` is misspelled.

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+        return m.edit(
+            new Discord.RichEmbed()
+                .setColor("#FF0000")
+                .addField("Failture", `Couldn't find a message with the id **${message_id}** in this server.`)
```
`failure` is misspelled.

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+                        return m.edit(
+                            new Discord.RichEmbed()
+                                .setColor("#FF0000")
+                                .addField("Failture", `The bots isn't the author of this message hence the bot can't edit this message.`)
```
`failure` is misspelled.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+                // await for the user's reply for 60 seconds.
+                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
+                    if (!collected.first()) {
+                        msg.edit(clean.setColor("#FF0000").addField("Failture", "Command Timed Out")).then(m => m.delete(30000))
```
`failure` is misspelled.

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+                    .setDescription("*This command will be automatically cancelled in 60 seconds*")
+                msg.edit(overwrite)
+                // await for the user's reply for 60 seconds.
+                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async function(collected) {
```

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+                            return false
+                        }
+                    }
+                }).catch(r => { msg.edit(clean.setColor("#FFFF00").addField("Information", "Command Cancelled.")).then(m => m.delete(30000)); console.error; return false })
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

Post-reformat, it's easier to see that there's something incorrect - `console.error` is mentioned with no function call, or arguments. Perhaps call it with `r` as an arg?
```suggestion
			}).catch(function(r) {
				msg.edit(clean.setColor("#FFFF00")
					.addField("Information", "Command Cancelled."))
					.then(m => m.delete(30000));
				console.error(r);
				return false;
			});
```

tyzoid commented on commands/addrole.js:
```diff
@@ -0,0 +1,127 @@
+        messageID: message_id
+    }, async (err, reaction_message) => {
+        msg = await message.reply("Loading...")
+        if (err) console.log(err)
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
	if (err)
		console.log(err)
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/clearroles.js:
```diff
@@ -0,0 +1,40 @@
+    }
+    reaction_message_model.deleteMany({
+        guildID: message.guild.id
+    },(err, rem) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
    }, function(err, rem) {
```

tyzoid commented on commands/editmessage.js:
```diff
@@ -0,0 +1,58 @@
+                                    `Message **${message_id}** have been edited.`
+                                )
+                        ).then(n=>n.delete(30000))
+                    }else{
```
Spaces around operators / control statements.
```suggestion
                    } else {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                        .addField("Failture", `Couldn't do it :cry:`)
+                    return message.channel.send(failture)
+                }
+                role_channel = message.guild.channels.find(f => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
                role_channel = message.guild.channels.find(function(f) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                role_channel = message.guild.channels.find(f => {
+                    if (f.id == rem.channelID) {
+                        return f
+                    }
```
What do you return here if `f.id != rem.channelID`? What's supposed to be returned? I doubt `undefined` is an expected return value here.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                        return f
+                    }
+                })
+                good_message = await role_channel.fetchMessage(message_id).catch(err => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

Newlines around large blocks are also nice.
```suggestion

                good_message = await role_channel.fetchMessage(message_id).catch(function(err) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                    console.log(err)
+                    return message.channel.send(embed)
+                })
+                reaction = await good_message.reactions.find(f => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion

                reaction = await good_message.reactions.find(function(f) {
```

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                reaction = await good_message.reactions.find(f => {
+                    if (f.me == true && f.emoji.name == reem) {
+                        return f
+                    }
```
What do you return here if `f.me != true` or `f.emoji.name != reem`? What's supposed to be returned? I doubt `undefined` is an expected return value here.

tyzoid commented on package.json:
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "discord.js": "^11.4.2",
     "luxon": "^1.11.4",
+    "mongoose": "^5.5.2",
```
Let's not import more dependencies until they're needed. Flatfiles should suffice for now.

tyzoid commented on commands/removerole.js:
```diff
@@ -0,0 +1,142 @@
+                })
+                if (reaction) {
+                    reaction.remove()
+                }
```
Let's put newlines around control blocks.

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+const Discord = require("discord.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
```
Move to flat file instead of jumping directly to a db.

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Shows all the roles on all the messages in the server
+ */
+module.exports = async (bot, message, args) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
module.exports = async function(bot, message, args) {
```

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+ */
+module.exports = async (bot, message, args) => {
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        message.reply("You don't have permission").then(async function(m) {
```

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+    // find all the roles that can be added with reacting on any message in this server
+    reaction_message_model.find({
+        guildID: message.guild.id
+    }, (err, records) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
    }, function(err, records) {
```

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+    reaction_message_model.find({
+        guildID: message.guild.id
+    }, (err, records) => {
+        if (err) { console.log(err); return }
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
	if (err)
		return console.log(err);
```

tyzoid commented on commands/roles.js:
```diff
@@ -0,0 +1,50 @@
+            return message.channel.send(norec).then(r => r.delete(60000))
+        }
+        // loop through all the found records, find the channel in which the message exists, then create a placeholder richembed
+        records.forEach(async record => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        records.forEach(async function(record) {
```

tyzoid commented on commands/tournaments.js:
```diff
@@ -24,4 +24,4 @@ function tournamentsCommand(msg, args) {
 }
 
-module.exports = tournamentsCommand;
+module.exports = tournamentsCommand;
```
No need to commit a file where the only change is the removal of a newline at the end of the file.

tyzoid commented on models/reaction_roles_message.js:
```diff
@@ -0,0 +1,13 @@
+const mongoose = require("mongoose");
```
Move to flat file instead of jumping directly to a db.

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+const Discord = require("discord.js")
+const mongoose = require('mongoose');
```
Move to flat file instead of jumping directly to a db.

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+ * @param {event} event raw event;
+ * bot.on("raw", event => {}
+ */
+module.exports.run = async (bot, event) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
module.exports.run = async function(bot, event) {
```

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+module.exports.run = async (bot, event) => {
+    var guild_obj = bot.guilds.find(r => r.id == event.d.guild_id)
+    var user_guild_member = await guild_obj.fetchMember(event.d.user_id).catch(err => console.log(err))
+    if(user_guild_member.user.bot) return
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
	if(user_guild_member.user.bot)
		return;
```

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+    var message_id = event.d.message_id
+    // fetch data from database
+    reaction_message_model.findOne({ messageID: message_id },
+        async (err, rec) => {
```
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        async function (err, rec) {
```

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+    // fetch data from database
+    reaction_message_model.findOne({ messageID: message_id },
+        async (err, rec) => {
+            if (err) console.log(err)
```
One-line if statements are against code standards, brackets may be omitted if only one-line, but the body line must appear on the next line.
```suggestion
		if (err)
			console.log(err)
```

tyzoid commented on reaction_roles.js:
```diff
@@ -0,0 +1,42 @@
+    reaction_message_model.findOne({ messageID: message_id },
+        async (err, rec) => {
+            if (err) console.log(err)
+            if (rec) {
```
Let's put newlines between control blocks.

tyzoid commented on utilities/hutil.js:
```diff
@@ -0,0 +1,50 @@
+module.exports = {
+    // return elements of an array in a string, side by side, separated by ', '
+    printarrayh(arrayy){
```
`array` is misspelled. Perhaps `arr` would be better?

tyzoid commented on utilities/hutil.js:
```diff
@@ -0,0 +1,50 @@
+        }
+        return output
+    }
+}
```
Not sure these functions are really all that useful or necessary here.

tyzoid reviewed (changes requested):
Overall, a decent start. A few things:

1. Replace spaces with tabs for indentation
2. Add semicolons where they should be. ASI is nice, but semicolons are for us human readers too.
3. Even though most of the creds have been removed, they still appear in commits prior to this one. These commits should be squashed / rebased together so as not to appear.

XtratoS committed:
Added semicolons

Added semicolons

XtratoS committed:
Added semicolons

Added semicolons

XtratoS committed:
Removed await

There's no need for await since we're returning the result anyways

XtratoS committed:
Removed await

There's no need for await since we're returning the result anyways

XtratoS committed:
Merge branch 'master' of https://github.com/XtratoS/backend

XtratoS committed:
Removed Unnecessary functions

Removed Unnecessary functions

.

XtratoS committed:
Modified according to last review

Changes:
-added semicolons.
-changed long functions from arrow notation to normal notation.
-fixed some indenation and formatting mistakes.
-args now has the command name as the first argument, also changed all command files accordingly, to start working with arguments from index 1 instead of 0.
-moved all requirments to the top of the files.

TODO:
-Test all the commands again.
-use json files pseudo-database instead of mongodb.

XtratoS committed:
Modified according to last review check desc.

Changes:
-added semicolons.
-changed long functions from arrow notation to normal notation.
-fixed some indenation and formatting mistakes.
-args now has the command name as the first argument, also changed all command files accordingly, to start working with arguments from index 1 instead of 0.
-moved all requirments to the top of the files.

TODO:
-use json files pseudo-database instead of mongodb

.

XtratoS committed:
Modified according to last review

Changes:
-added semicolons.
-changed long functions from arrow notation to normal notation.
-fixed some indenation and formatting mistakes.
-args now has the command name as the first argument, also changed all command files accordingly, to start working with arguments from index 1 instead of 0.
-moved all requirments to the top of the files.

TODO:
-use json files pseudo-database instead of mongodb.

XtratoS committed:
Merge branch 'master' of https://github.com/XtratoS/backend

XtratoS committed:
Merge branch 'master' of https://github.com/XtratoS/backend

XtratoS committed:
Modified syntax for multiple command files

*Modified the code for better readability (improved indentation and spacing)
*Better variable initilization and usage (for ex. using a single placeholder if a variable of the same type is used multiple times in the same file before returning)

XtratoS committed:
Added logging bases

Added the code for logging an action, a logchannel command that selects the logchannel, and a file based pesudodatabase that stores the logging channel to send the logs there everytime a logging action is taken

XtratoS committed:
Event listener for users joining the server

Added an event listener for users joining the server that sends them a message when they do, if they don't succeed we can take a different action, also logs the action to the registered logging channel if exists

XtratoS committed:
Add documentation

Added function documentation

XtratoS committed:
Added semicolons

Added semicolons

XtratoS committed:
Added an event listener for role add/remove

Added an event listener for role add/remove with logging for every role addition/removal action

XtratoS committed:
Bug fix and code optimization

- Using args[1] instead of args[0] for finding the first argument ( bug fix )
- Used a more simple function for finding the channel in the guild

XtratoS committed:
Added a raw event listener

Added a raw event listener

XtratoS committed:
Moved reaction roles file

Moved reaction roles file to suitable directory and updated required file calls

XtratoS committed:
Code optimization/organization

Moved the raw event listener code to a separate file, and added role update listener and user join listener from separate files aswell

XtratoS committed:
added nickname change logger

Changed the role_update to member_update to listen to all changes to a user, currently listens to role and nickname changes

XtratoS committed:
Create message.js

XtratoS committed:
Added moveall command

Added moveall command to move all users in any of the voice channels in the server to a specified voice channel by its name/id

XtratoS committed:
Update move_voice_all.js

XtratoS committed:
Update addrole.js

XtratoS committed:
Update clearroles.js

XtratoS committed:
Add user_leave event listener

Add user_leave event listener

XtratoS committed:
Added missing requirements

Added missing requirements

XtratoS committed:
Added the addnote command

Added the addnote command to add a note to a user ( uses pseudo/file database )

XtratoS committed:
Added viewnotes command

Added viewnotes command to view all the notes for a specified user

XtratoS committed:
Autoformatted the file with VSCode

Autoformatted the file with VSCode

XtratoS committed:
Update log_channel.js

Replaced arrow function notation with normal one

XtratoS committed:
Added user_leave and message event listeners

Added user_leave and message event listeners

XtratoS committed:
Define the new commands

Define the new commands

XtratoS committed:
Autoformat

Autoformat VSCode

XtratoS committed:
Update addnote.js

XtratoS committed:
Update log_channel.js

XtratoS committed:
Update roles.js

XtratoS committed:
Update package.json

XtratoS committed:
Adding Date/Time to logging

Adding Date/Time to logging for join/leave events

XtratoS committed:
Create add_log_channel.js

Added addlogchannel command which adds the logging channel for several events

XtratoS committed:
Added mute command

Added mute command, works as tempmute if a timeframe is given

XtratoS committed:
Add unmute command

Add unmute command

XtratoS committed:
Update removerole.js

XtratoS committed:
Update makemessage.js

XtratoS committed:
Update log.js

XtratoS committed:
Update commands.js

XtratoS committed:
Update roles.js

XtratoS committed:
Moved to actions folder

Moved to actions folder

XtratoS committed:
Fixing some issues with addnote/viewnote commands

Fixing minor issues with addnote/viewnote commands

XtratoS committed:
Fixed member update event listener

Fixed member update event listener ( logs role updates and nickname changes )

XtratoS committed:
Added ban, unban, softban and welcome_message commands

Added ban, unban, softban and welcome_message commands

XtratoS committed:
Modified/fixed addnote and viewnotes commands

Modified/fixed addnote and viewnotes commands

XtratoS committed:
Added more logging options

Added more logging channels options for the new commands, also added the "all" option which adds the logging channel to log all the events in the same channel, channels can be overwritten by adding them again

XtratoS committed:
Allowing enabling/disabling the welcome message aswell as easily modifiying it through commands

Allowing enabling/disabling the welcome message aswell as easily modifiying it through commands

XtratoS committed:
Added the welcome_message command

Added the welcome_message command, which allows you to disable the welcome_message or enable it for the bot to send the welcome message you specify to all members that join the server

XtratoS committed:
Delete log_c.txt

XtratoS committed:
Ignore all db files

Ignore all db files

XtratoS committed:
Add a semicolon

XtratoS committed:
Add ban command

Add ban command to ban members permanently from the server

XtratoS committed:
Bug fixes for editmessage command

Bug fixes for editmessage command

XtratoS committed:
Added eval command to run a piece of code inside the discord chat and show the result

Added eval command to run a piece of code inside the discord chat and show the result - restricted to ADMINISTRATOR permissions, be very careful when using this

XtratoS committed:
Bug fix

Bug fix

XtratoS committed:
Date/Time now shows in the footer of the log

Date/Time now shows in the footer of the log

XtratoS committed:
bug fix for removerole command

bug fix for removerole command

XtratoS committed:
Added softban command to ban a user, deleting his messages for the past 7 days, then unbanning the user and sending him an invite back to the server

Added softban command to ban a user, deleting his messages for the past 7 days, then unbanning the user and sending him an invite back to the server

XtratoS committed:
Added unban command to unban a banned user

Added unban command to unban a banned user

XtratoS committed:
Bug fixes for user_leave event listener

Bug fixes for user_leave event listener

XtratoS committed:
Date/Time now shows in the footer of the log embed

Date/Time now shows in the footer of the log embed
